### PR TITLE
Fix bug in input parser

### PR DESCRIPTION
**Pull request by mbadams5**
See https://unmethours.com/question/34222/error-parsing-schedulefile-error-in-following-line-in-e-89/

Strings that had \r, \n, and other special characters were being parsed as true carriage return, newline, etc. This was causing issues with file paths on windows where they could legitimately have \r or \n in the file path.

Pull request overview
---------------------
Please change this line to a description of the pull request, with useful supporting information including whether it is a new feature, or fixes a defect, a cross reference to any defects addressed in this PR, the type of changes to be made, and whether diffs are expected/justified based on this change.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: tst/EnergyPlus/unit/InputProcessor.unit.cc [hunk 1]
```diff
@@ -1407,16 +1458,16 @@ TEST_F(InputProcessorFixture, parse_string)
     index = 0;
     success = true;
     output_string = parse_string(R"(\b\t/\\\";)", index, success);
-    EXPECT_EQ("\b\t/\\\"", output_string);
+    EXPECT_EQ(R"(\b\t/\\\")", output_string);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/InputProcessor.unit.cc:1461]: > `R"(\b\t/\\\")"`

image

File: src/EnergyPlus/InputProcessing/IdfParser.cc [hunk 2]
```diff
@@ -572,30 +572,6 @@ std::string IdfParser::parse_string(std::string const &idf, size_t &index, bool
             complete = true;
             decrement_both_index(index, index_into_cur_line);
             break;
-        } else if (c == '\\') {
```
Comment (Myoldmopar) [inline src/EnergyPlus/InputProcessing/IdfParser.cc:575]: So really it's just removing this code to fix it?  That's a nice thing...

File: tst/EnergyPlus/unit/InputProcessor.unit.cc [hunk 3]
```diff
@@ -284,6 +284,57 @@ TEST_F(InputProcessorFixture, decode_encode_2)
...
+      "  8760.0,",
+      "  Comma,",
+      "  ,",
+      "  10.0;",
+      ""
+    }));
+
+    ASSERT_TRUE(process_idf(idf));
+    std::string encoded = encodeIDF();
+    EXPECT_EQ(expected, encoded);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/InputProcessor.unit.cc:335]: Ahhh this was confusing for a minute.  I forgot the test fixture is going to add the building and global geometry objects automatically.  Ok, ok.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
CI tests look OK, code is interesting.  @mbadams5 could you describe why removing that block of code fixes it?  I mean I can imagine the parser simply moves into a different block and leaves `\r` alone, but can you imagine any consequences?  I feel like you added that code on purpose when you wrote the parser originally.

**Comment by mbadams5** [comment]
@Myoldmopar 

So what was happening is that with that section of code, when a string was parsed and it finds a `\` then it looks to see if it is an escaped special character (`\"`, `\r`, `\n`, etc). If it finds one of these special characters it then turns it into that special character, meaning instead of 2 `char` (`\` and `t`) it becomes `\t`, the tab character. This kind of functionality is important when parsing formats such as JSON where you can have a string with `\n` in it and want the string when parsed to have a newline but NOT the JSON file itself. That said, I don't think this is appropriate when parsing IDF and is in fact causing this bug since IDF doesn't have explicit string delimited fields, `""`. This portion of the code was adapted from some prior work and shouldn't have been added in the first place for IDF parsing.

Hopefully this explanation was clear as mud.

**Comment by Myoldmopar** [comment]
That's a perfect explanation.  I am good with the code changes here.  I'll try to play around with it to verify for a few minutes this morning and if it's all good I'll go ahead and merge.

**Comment by Myoldmopar** [comment]
Tested this in develop by putting in some escape characters in a test file.  It fails with an explanatory error message.  Running this in this branch allows it to run just fine.  Merging.  Thanks @mbadams5.

**Pull request merged**